itest/utils.go

```diff
@@ -735,6 +739,182 @@ func ManualMintSimpleAsset(t *harnessTest, lndNode *node.HarnessNode,
+
+	// Extract group witness from signed PSBTs.
+	var groupWitnesses []*taprpc.GroupWitness
+	for idx := range callbackRes {
+		res := callbackRes[idx]
+		signedPsbt := res.SignedPsbt
+		genesisAssetID := res.AssetID
+
+		// Sanity check signed PSBT.
+		require.Len(t.t, signedPsbt.Inputs, 1)
+		require.Len(t.t, signedPsbt.Outputs, 1)
+
+		// Extract witness from signed PSBT.
+		witnessStack, err := DeserializeWitnessStack(
+			signedPsbt.Inputs[0].FinalScriptWitness,
+		)
+		if err != nil {
+			log.Fatalf("Failed to deserialize witness stack: %v",
+				err)
+		}
+
+		groupWitness := taprpc.GroupWitness{
+			GenesisId: genesisAssetID[:],
+			Witness:   witnessStack,
+		}
+
+		groupWitnesses = append(groupWitnesses, &groupWitness)
+	}
+
+	// Seal the batch with the group witnesses.
+	ctxt, cancel = context.WithTimeout(ctxb, defaultWaitTimeout)
+
+	sealReq := mintrpc.SealBatchRequest{
+		GroupWitnesses: groupWitnesses,
+	}
+	sealResp, err := tapNode.SealBatch(ctxt, &sealReq)
+	require.NoError(t.t, err)
+
+	// Cancel the context for the seal request call.
+	cancel()
+
+	require.NotEmpty(t.t, sealResp.Batch)
+
+	// With the batch sealed successfully, we can now finalize it and
+	// broadcast the anchor TX.
+	ctxt, cancel = context.WithCancel(context.Background())
+	defer cancel()
+	stream, err := tapNode.SubscribeMintEvents(
+		ctxt, &mintrpc.SubscribeMintEventsRequest{},
+	)
+	require.NoError(t.t, err)
+	sub := &EventSubscription[*mintrpc.MintEvent]{
+		ClientEventStream: stream,
+		Cancel:            cancel,
+	}
+
+	batchTXID, batchKey := FinalizeBatchUnconfirmed(
+		t.t, t.lndHarness.Miner().Client, tapNode, assetReqs,
+	)
+	batchAssets := ConfirmBatch(
+		t.t, t.lndHarness.Miner().Client, tapNode, assetReqs, sub,
+		batchTXID, batchKey,
+	)
+
+	return batchAssets
+}
+
+// DeserializeWitnessStack deserializes a serialized witness stack into a
+// [][]byte.
+//
+// TODO(ffranr): Do we already have this function somewhere else? Move to btcd?
+func DeserializeWitnessStack(serialized []byte) ([][]byte, error) {
+	var stack [][]byte
+	reader := bytes.NewReader(serialized)
+
+	// Read the number of witness elements (compact size integer)
+	count, err := readCompactSize(reader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read witness count: %w", err)
+	}
+
+	// Read each witness element
+	for i := uint64(0); i < count; i++ {
+		elementSize, err := readCompactSize(reader)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read witness "+
+				"element size: %w", err)
+		}
+
+		// Read the witness element data
+		element := make([]byte, elementSize)
+		if _, err := reader.Read(element); err != nil {
+			return nil, fmt.Errorf("failed to read witness "+
+				"element data: %w", err)
+		}
+		stack = append(stack, element)
+	}
+
+	return stack, nil
+}
+
+// readCompactSize reads a compact size integer from the given reader.
+func readCompactSize(r *bytes.Reader) (uint64, error) {
```

Comment:
I think you can use `ReadVarInt` here. See https://pkg.go.dev/github.com/btcsuite/btcd/wire#ReadVarInt